ReactWindows/ReactNative.Shared/UIManager/IViewManagerExtension.cs

```diff
@@ -0,0 +1,81 @@
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+#if WINDOWS_UWP
+using Windows.UI.Xaml;
+#else
+using System.Windows;
+#endif
+
+namespace ReactNative.UIManager
+{
+    /// <summary>
+    /// An interface for extending view managers with additional props.
+    /// </summary>
+    public interface IViewManagerExtension
+    {
+        /// <summary>
+        /// The type of view manager type being extended.
+        /// </summary>
+        Type ViewManagerType { get; }
+
+        /// <summary>
+        /// The commands map for the view manager extension.
+        /// </summary>
+        IReadOnlyDictionary<string, object> CommandsMap { get; }
+
+        /// <summary>
+        /// The exported custom bubbling event types.
+        /// </summary>
+        IReadOnlyDictionary<string, object> ExportedCustomBubblingEventTypeConstants { get; }
+
+        /// <summary>
+        /// The exported custom direct event types.
+        /// </summary>
+        IReadOnlyDictionary<string, object> ExportedCustomDirectEventTypeConstants { get; }
+
+        /// <summary>
+        /// The exported view constants.
+        /// </summary>
+        IReadOnlyDictionary<string, object> ExportedViewConstants { get; }
+
+        /// <summary>
+        /// The native props for the view manager extension.
+        /// </summary>
+        IReadOnlyDictionary<string, string> NativeProperties { get; }
+
+        /// <summary>
+        /// Called when the view is created.
+        /// </summary>
+        /// <param name="reactContext">The context.</param>
+        /// <param name="view">The view.</param>
+        void OnCreateView(ThemedReactContext reactContext, DependencyObject view);
```

Comment:
can DependencyObject be parameterized <T> to make this interface not XAML-specific?